src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java

```diff
@@ -517,7 +517,7 @@ public String nativeSQL(String sql) throws SQLException {
 	@Override
 	public boolean isReadOnly() throws SQLException {
 		validateConnectionIsNotClose();
-		return false;
+		return sessionProperties.isSystemEngine();
```

Comment:
Can we claim read-only if we can create tables via system engine? Should we?

Reply:
hmm. you are right. Let's return false.